app/decorators/user_decorator.rb

```diff
@@ -20,17 +20,47 @@ class UserDecorator < ApplicationDecorator
 
   DEFAULT_PROFILE_SUMMARY = "404 bio not found".freeze
 
+  # The relevant attribute names for cached tags.  These are the attributes that we'll make
+  # available in the front-end.  The list comes from the two places (see below for that list).
+  #
+  # @see app/controllers/async_info_controller.rb
+  # @see app/services/articles/feeds/article_score_calculator_for_user.rb
+  CACHED_TAGGED_BY_USER_ATTRIBUTES = %i[bg_color_hex hotness_score id name points text_color_hex].freeze
+
+  # A proxy for a Tag object.  In app/services/articles/feeds/article_score_calculator_for_user.rb
+  # we rely on method calls to the object.  (e.g. "tag.name").  This class helps us conform to that
+  # expectation.
+  #
+  # @note A Struct in Rails can be cast "to_json" and uses it's attributes.
+  #
+  # @see https://github.com/rails/rails/blob/main/activesupport/lib/active_support/core_ext/object/json.rb#L68-L72
+  CachedTagByUser = Struct.new(*CACHED_TAGGED_BY_USER_ATTRIBUTES) do
```

Comment:
I think we can simplify this by using `keyword_init: true`:

![Screen Shot 2022-01-24 at 09 37 13](https://user-images.githubusercontent.com/47985/150713001-4319bd86-8c27-4744-96e9-37ed4eceab50.png)

As you can see in the above example it works with string and symbol keys, so the entire `from_hash` method can either be skipped entirely (and we use `new` directly)  or shortened to this:

```ruby
def self.from_hash(hash)
  new(hash)
end
```

Reply:
Today I learned about `keyword_init: true`